docs/guides/latest/volumesnapshot/deployment-backup.md

```diff
@@ -0,0 +1,262 @@
+
+Sample `BackupConfiguration` crd YAML are given below,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: deployments-volume-snapshot
+  namespace: demo
+spec:
+  schedule: "*\1 * * * *"
+  driver: VolumeSnapshotter
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: Deployment
+      name: stash-demo
+    snapshotClassName: default-snapshot-class
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Here,
+* `spec.schedule` is a [cron expression](https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#schedule) that indicates that `BackupSession` will be created every 1 minute. Then `BackupSession` crd will start taking the periodic backup of the data.
+
+* `spec.driver` The driver indicates the name of the agent to use to back up the target. Supported values are `Restic`, `VolumeSnapshotter`. `VolumeSnapshotter` are used for creating/restoring volume snapshot.
+
+* `spec.target.ref`  refers to the backup target. `apiVersion`, `kind` and `name` refers to the `apiVersion`, `kind` and `name` of the targeted workload respectively. Stash will use this information to create a Volume Snapshotter Job for Creating Volume Snapshot.
+
+* `spec.target.snapshotClassName` indicates the [VolumeSnapshotClass](https://kubernetes.io/docs/concepts/storage/volume-snapshot-classes/) to use for volume snapshotting.
+
+
+Let's create the `BackupConfiguration` crd we have shown above.
+
+
+
+
+```console
+$ kubectl apply -f ./docs/examples/volume-snapshot/backupconfiguration.yaml
+backupconfiguration.stash.appscode.com/deployments-volume-snapshot created
+```
+here, `deployments-volume-snapshot` is the name of the `BackupConfiguration` crd that has been created successfully.
+
+##### Verify Creating Volume Snapshot 
+
+If everything goes well, the Stash operator will create a `BackupSession` crd with the CronJob schedule. `BackupSession` crd name will be automatically generated by the following pattern `<BackupConfiguration name>-<creation timestamp in Unix epoch seconds>` and Label should be set as the following pattern `backup-configuration:<BackupConfiguration name>`.
+
+Now, wait a few minutes,  you will see that `BackupSession` has been created.
+Check that the phase of `BackupSession` status is `Running` 
+
+```console
+$ kubectl get bs -n demo -l  backup-configuration=deployments-volume-snapshot
+NAME                                     BACKUPCONFIGURATION           PHASE       AGE
+deployments-volume-snapshot-1559026686   deployments-volume-snapshot   Running   26s
+```
+here, `deployments-volume-snapshot-1559026686` is the name of the `BackupSession` crd that has been created successfully after 1 minute.
+
+
+`BackupSession` crd will create volume snapshotter job, job name follows the following pattern `volume-snapshot-<BackupSession name>`. 
```

Comment:
Stash operator watches for `BackupSession` crd and it will create a volume snapshotter job named `volume-snapshot-<BackupSession name>`.